### PR TITLE
MP-2066 Generic shipment request validation

### PR DESCRIPTION
**Pull request by NickVries**
**Related Issues**
- https://myparcelcombv.atlassian.net/browse/MP-2066

**Notes**
- Apparently the FoundationServiceProvider is necessary for FormRequests to work, so we added it.
- The redirect property on the FormRequest class prevents laravel from trying to interact with a session that isn't there.

### CODE REVIEW

File: app/Http/FormRequest.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+declare(strict_types=1);
+
+namespace MyParcelCom\Microservice\Http;
+
+use Illuminate\Foundation\Http\FormRequest as BaseFormRequest;
+
+class FormRequest extends BaseFormRequest
+{
+    protected $redirect = '/shipments';
```
Comment (M4tini) [inline app/Http/FormRequest.php]: Can you add a docblock mentioning this part of the PR description:
> prevents laravel from trying to interact with a session that isn't there

File: app/Http/FormRequest.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * @return bool
+     */
+    public function authorize(): bool
```
[the remark this answers is not in the document]
Comment (NickVries) [reply app/Http/FormRequest.php]: Unfortunately, yes. This defaults to false, which causes the microservice to return with status 403. 

From the Laravel documentation:
image

File: app/Http/ShipmentRequest.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+     * ]
+     * This will enforce the description attribute to be set and to be a string
+     * and that the weight of the shipment is an integer and is at least 1000.
+     *
+     * See the laravel documentation for all available validation rules:
+     * https://laravel.com/docs/5.5/validation#available-validation-rules
+     *
+     * @return array
+     */
+    protected function carrierSpecificShipmentRules(): array
```
[the remark this answers is not in the document]
Comment (NickVries) [reply app/Http/ShipmentRequest.php]: We can call it anything you like :)

[the remark this answers is not in the document]
Comment (nessy) [reply app/Http/ShipmentRequest.php]: So.....

**Pull request merged**